scope.go

```diff
@@ -500,19 +502,54 @@ func (s *scope) Snapshot() Snapshot {
 }
 
 func (s *scope) Close() error {
+	// n.b. Once this flag is set, the next scope report will remove it from
+	//      the registry and clear its metrics.
 	if !s.closed.CAS(false, true) {
 		return nil
 	}
 
 	close(s.done)
-	s.reportRegistry()
 
-	if closer, ok := s.baseReporter.(io.Closer); ok {
-		return closer.Close()
+	if s.root {
+		s.reportRegistry()
+		if closer, ok := s.baseReporter.(io.Closer); ok {
+			return closer.Close()
+		}
 	}
+
 	return nil
 }
 
+func (s *scope) clearMetrics() {
+	s.cm.Lock()
+	s.gm.Lock()
+	s.tm.Lock()
+	s.hm.Lock()
+	defer s.cm.Unlock()
+	defer s.gm.Unlock()
+	defer s.tm.Unlock()
+	defer s.hm.Unlock()
+
+	for k := range s.counters {
```

Comment:
Instead of iterating, why not `s.counter = make(map[string]struct{})` and let go runtime garbage-collect the references?